src/governance/governance.cpp

```diff
@@ -342,15 +342,14 @@ void CGovernanceManager::AddGovernanceObject(CGovernanceObject& govobj, CConnman
     GetMainSignals().NotifyGovernanceObject(std::make_shared<const Governance::Object>(govobj.Object()));
 }
 
-void CGovernanceManager::UpdateCachesAndClean()
+void CGovernanceManager::CheckAndRemove()
 {
-    // Return if meta manager hasn't had a chance to load its database yet
-    if (!::mmetaman->IsValid()) return;
+    assert(::mmetaman->IsValid());
```

Comment:
Because earlier, `CheckAndRemove`, a function meant for cleaning up database entities, is called _regardless_ of the database being present or not (case and point in `CDSNotificationInterface::NotifyMasternodeListChanged`, whereas now it will check if governance's database is valid before trying to clean its contents, [source](https://github.com/dashpay/dash/pull/5966/commits/594312e392eb2da67186d0188811208739474a7d#diff-445b6e7d79b7494974867b78734da7ca26a26ae21fd687c16c20d0910cf07c33R134-R140))

Another thing was that governance's db was being initialized (and cleaned up, [source](https://github.com/dashpay/dash/blob/594312e392eb2da67186d0188811208739474a7d/src/governance/governance.cpp#L70)) _before_ its dependencies `netfulfilledman` and `mn_metaman` dbs can load, that has also been corrected ([source](https://github.com/dashpay/dash/pull/5966/commits/594312e392eb2da67186d0188811208739474a7d#diff-b1e19192258d83199d8adaa5ac31f067af98f63554bfdd679bd8e8073815e69dR2218-R2242)).

As we've gotten rid of avenues where `CheckAndRemove` can be called without a valid database, we can now replace the fast-return with an assertion to enforce the order of database initialization (first `netfulfilledman`, then `mn_metaman`, then `governance`)